whelktool/scripts/cleanups/2020/06/lxl-3187-tgm-relink-saogf-links.groovy

```diff
@@ -0,0 +1,59 @@
+/**
+ *
+ * See LXL-3187 for more information
+ */
+
+report = getReportWriter("report.txt")
+gmgpcPrefixSwe = "https://id.kb.se/term/gmgpc/swe/"
+saogfPrefix = "https://id.kb.se/term/saogf/"
+
+selectByCollection('auth') { data ->
+    def instance = data.graph[1]
+
+    if (!instance?.inScheme?.'@id'?.contains("https://id.kb.se/term/gmgpc")) {
+        return
+    }
+
+    def relations = ['broader', 'narrower', 'related']
+
+    boolean updated = false
+
+    relations.each { relation ->
+        if (!instance[relation]) {
+            return
+        }
+
+        //Objects containing @type, prefLabel and sameAs
+        def toRemove = instance[relation].findAll{ hasSaogfSameAs(it) }
+        def toAdd  = toRemove.collect{ ['@id' : gmgpcPrefixSwe + it.sameAs[0][ID].tokenize('/').last()] }
```

Comment:
It may be good to actually verify that the newly created uri exists. PrefLabel is created manually and may contain errors, although the terms should be considered stable. But if not verify then maybe at least log which ones are created? Or test it thoroughly!